frontend/packages/console-shared/src/utils/validation/validation.ts

```diff
@@ -0,0 +1,102 @@
+import { addMissingSubject, joinGrammaticallyListOfItems, makeSentence } from '../grammar';
+import { ValidationErrorType, ValidationObject } from './types';
+import {
+  DNS1123_END_ERROR,
+  DNS1123_START_END_ERROR,
+  DNS1123_START_ERROR,
+  DNS1123_TOO_LONG_ERROR,
+  EMPTY_ERROR,
+} from './strings';
+
+const alphanumericRegex = '[a-zA-Z0-9]';
+
+const DNS_1123_OFFENDING_CHARACTERS = {
+  ',': 'comma',
+  "'": 'apostrophe', // eslint-disable-line quotes
+  _: 'underscore',
+};
+
+export const getValidationObject = (
+  message: string,
+  type: ValidationErrorType = ValidationErrorType.Error,
+): ValidationObject => ({
+  message,
+  type,
+});
+
+export const validateEmptyValue = (
+  value: string,
+  { subject } = { subject: undefined },
+): ValidationObject => {
+  if (!value) {
+    return getValidationObject(
+      addMissingSubject(makeSentence(EMPTY_ERROR, false), subject),
+      ValidationErrorType.TrivialError,
+    );
+  }
+  return null;
+};
+
+// DNS-1123 subdomain
+export const validateDNS1123SubdomainValue = (
+  value: string,
+  { subject } = { subject: undefined },
```

Comment:
not sure if we should keep this opts `{}` api  when it gets public now. There is just one option and it is a bit harder to write. It was meant to be extensible but not sure if it is really necessary. 

We could also add `Name` as a default value of the subject or simply make the subject required

Reply:
If we remove it, I would suggest doing it in another PR since it will require changes in more than one place. I'm comfortable with either adding `Name` as the default value or making it required, though. I think I'd lean more towards providing it a default value to keep the API simple, but still flexible.

Reply:
Done! I add `Name` as the default value.